src/caching/IPFSClient.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  buildIPFSClient,
+  jsonReplacerWithBigNumbers,
+  jsonReviverWithBigNumbers,
+  retrieveValueFromIPFS,
+  storeValueInIPFS,
+} from "../utils/IPFSUtils";
+import { CachingMechanismInterface } from "../interfaces";
+import { Struct } from "superstruct";
+import winston from "winston";
+import PinataClient from "@pinata/sdk";
+
+/**
+ * A client for interacting with the IPFS. This is a wrapper around the IPFS API.
+ * This client also is a part of the caching mechanism interface.
+ * @see https://docs.ipfs.io/reference/http/api/
+ */
+export class IPFSClient implements CachingMechanismInterface {
+  /**
+   * The IPFS client instance
+   */
+  private client: PinataClient;
+
+  /**
+   * Public IPFS gateway URL to retrieve the content from
+   */
+  private publicGatewayURL: string;
+
+  /**
+   * An optional logger for logging messages
+   */
+  private logger?: winston.Logger;
+
+  public constructor(projectId: string, projectSecret: string, publicGatewayURL: string, logger?: winston.Logger) {
+    this.client = buildIPFSClient(projectId, projectSecret);
+    this.publicGatewayURL = publicGatewayURL;
+    this.logger = logger;
+  }
+
+  /**
+   * Calls to a public IPFS gateway to retrieve a value.
+   * @param key The key to retrieve.
+   * @param _structValidator An optional struct validator to validate the retrieved value. If the value does not match the struct, null is returned.
+   * @returns The value if it exists, otherwise null.
+   */
+  async get<ObjectType>(key?: string, _structValidator?: Struct<unknown, unknown>): Promise<ObjectType | null> {
+    if (!key) {
+      return null;
+    }
+    this.logger?.info(`Retrieving value from IPFS with key ${key}`);
```

Comment:
This logging doesn't comply with our existing log format as it should include the following keys:
- `at`: i.e. "IPFSClient#get"
- `message`: i.e. `Retrieving value from IPFS with key ${key}`. 
- ...other props you want to log

Reply:
Done & included a new formatting fn so that the logger is typed. https://github.com/across-protocol/sdk-v2/pull/341/commits/18bc951b56ddad980fd167bf86e29e72806f958e